promql/parse.go

```diff
@@ -665,6 +660,9 @@ func (p *parser) primaryExpr() Expr {
 	case t.typ.isAggregator():
 		p.backup()
 		return p.aggrExpr()
+
+	default:
+		p.errorf("no valid expression found")
 	}
 	return nil
```

Comment:
Only if I move `return nil` into the default. Which is also never executed as `p.errorf` panics.
Both options are equally noops.